### DISCUSSION TITLE
Undo/Redo functionality not working consistently with collaboration

### DISCUSSION THREAD

**Discussion by GSKarthik369** [ueberdosis/tiptap#4978] [General Questions]
Undo/Redo are not working consistently when we are using the collaborative editor feature in Tiptap Reactjs

It doesn't work either with **Ctrl+Z** or **Ctrl+Y(Ctrl + Shift + Z)**

OR

even when we call [undo functions](https://tiptap.dev/docs/editor/api/extensions/collaboration#undo) or [redo functions](https://tiptap.dev/docs/editor/api/extensions/collaboration#redo)

OR even using the functions below

**Undo**
` editor.chain().focus().undo().run();`

**Redo**
` editor.chain().focus().redo().run();`

On the first load of Tiptap reactjs editor it shows a transcation change like below with all the nodes in the steps object
![History](https://github.com/ueberdosis/tiptap/assets/34792988/bcc8fed0-efcb-44d9-810e-5036c6cc2189)

This steps of all the nodes also comes when we randomly do undo and redo even while using **Ctrl + Z** or even the [undo functions](https://tiptap.dev/docs/editor/api/extensions/collaboration#undo) call.

At first when the editor instance was created we were doing some mistakes like calling the Collaboration and default History extensions together

 Like below
 
 ```
   const editor = useEditor({
    editable: false,
    extensions: [
          StarterKit.configure({
              history: false,
           }),
         History,
         Collaboration,
         CollaborationCursor,
         **add so on**
    ],
    enableInputRules: false,
    enablePasteRules: false,
    onCreate({ editor }) {
       Functions been called here
    },
    onUpdate({ editor }) {
        **Functions been called here**
    },
    onDestroy() {
        **Functions been called here**
    },
    onTransaction({ editor }) {
        **Functions been called here**
    },
    onSelectionUpdate({ editor }) {
        **Functions been called here**
    },

    editorProps: {
      attributes: {
        spellcheck: "false",
      },
    },
    autofocus: "start",
  });
 ```
 The above issue was resolved by removing the default History extension plugin from the editor instance after refering this [Documentation](https://tiptap.dev/docs/editor/api/extensions/collaboration)
 
We also have trackchanges extension implemented in our Tiptap reactjs editor from this [Trackchanges Extension](https://track-change.onrender.com/)

In the above trackchanges extension reusage we also have similar functions like onTransaction. If that is causing the issue of giving this all nodes in the steps while doing each transaction

Any help would be very grateful?

@bdbch @janthurau

**Comment by sukets16** [comment]
+1 I'm also facing this issue about a year out.
